torch/_tensor.py

```diff
@@ -1012,6 +1012,34 @@ def __torch_function__(cls, func, types, args=(), kwargs=None):
             ret = func(*args, **kwargs)
             return _convert(ret, cls)
 
+    def __dlpack__(self, stream=None):
+        """
+        Creates a dlpack capsule of the current tensor to
+        be exported to other libraries.
+
+        This function will be called from the `from_dlpack` method
+        of the consumer library.
+
+        Args:
+            stream (integer or None): Object that represents a stream and provides
+                a `synchronize` method. Optional.
+        """
+        if isinstance(stream, torch.cuda.Stream) or hasattr(stream, 'synchronize'):
+            stream.synchronize()
```

Comment:
This shouldn't be necessary, but maybe the right API is missing at the Python level here to do this asynchronously? From https://github.com/dmlc/dlpack/issues/57#issuecomment-753696812:

_"In cases where both sides uses their own stream, async exchange can still be done by stream dependency queing"_:
```c
// event can also be created on the fly, or create a synchronizer object and cache it.
// We could build auxiliary function that can be called from python side if that helps the frameworks
void PushStreamDep(cudaStream_t src, cudaStream dst) {
    cudaEvent_t event;
    cudaEventCreate(&event);
    cudaEventRecord(&event ,src);
    cudaStreamWaitForEvent(dst, event);
   cudaEventDestroy(&event);
}
```